### PR TITLE
move global trace provider api to global package.

### PR DESCRIPTION
**Pull request by rghetia**
partially addresses #216

### CODE REVIEW

File: global/global.go [hunk 1]
```diff
@@ -12,28 +12,32 @@
...
 type globalProvider struct {
-	p Provider
+	p trace.Provider
 }
 
 var globalP atomic.Value
 
-// GlobalProvider returns trace provider registered with global registry.
-// If no trace provider is registered then an instance of NoopTraceProvider is returned.
+// TraceProvider returns registered global trace provider.
```
Comment (cbandy) [inline global/global.go]:
```suggestion
// TraceProvider returns the registered global trace provider.
```
or
```suggestion
// TraceProvider returns the configured global trace provider.
```

File: global/global.go [hunk 2]
```diff
@@ -12,28 +12,32 @@
...
 		return gp.(globalProvider).p
 	}
-	return NoopTraceProvider{}
+	return trace.NoopTraceProvider{}
 }
 
-// SetGlobalProvider sets the provider as a global trace provider.
-func SetGlobalProvider(m Provider) {
-	globalP.Store(globalProvider{p: m})
+// SetTraceProvider configures the provider as a global trace provider.
```
Comment (cbandy) [inline global/global.go]:
```suggestion
// SetTraceProvider registers p as the global trace provider.
```
or
```suggestion
// SetTraceProvider configures p to be the global trace provider.
```

**Comment by jmacd** [comment]
There is a static function in `api/trace/api.go`,

```
// Start starts a new span using registered global tracer.
func Start(ctx context.Context, name string, opts ...SpanOption) (context.Context, Span) {
	return GlobalTracer().Start(ctx, name, opts...)
}
```

I think this should be eliminated. There should be no uses of a global from within the `api/trace` package i9tself.

File: global/global.go (hunk 1 above)
Comment (rghetia) [reply global/global.go]: done.

File: global/global.go (hunk 2 above)
Comment (rghetia) [reply global/global.go]: done.

**Comment by rghetia** [comment]
> I think this should be eliminated. There should be no uses of a global from within the `api/trace` package i9tself.

It was eliminated in #227

**Pull request merged**